portfolio/src/main/webapp/pagination.js

```diff
@@ -0,0 +1,185 @@
+
+document.addEventListener('DOMContentLoaded', () => {
+  const commentButton = new CommentButton();
+  commentButton.requestAndDisplayNewCommentsPage();
+});
+
+class CommentButton {
+  constructor() {
+    this.currentCommentsPageIndex = 0;
+    this.totalNumberOfPagesRequested = 1;
+    this.lastPage = -1;
+    this.cursorString = null;
+    this.prevCommentsButton = document.getElementById('prev-comments-button');
+    this.nextCommentsButton = document.getElementById('next-comments-button');
+    this.endOfCommentsSign = document.getElementById('end-of-comments-sign');
+    this.comments = document.getElementById('comments');
+
+    this.prevCommentsButton.classList.add("hidden");
+    this.endOfCommentsSign.classList.add("hidden");
+   
+    document.getElementById('maxNum').addEventListener('input', () => {
+      // Resetting innerHTML of comments when user re-selects the number of maximum comments to display
+      this.comments.innerHTML = "";
+      // Displays most recent page of comments so "prev" button should be hidden, "end-of-comments-sign" should be hidden, while "next" button should display.
+      this.prevCommentsButton.classList.add("hidden");
+      this.nextCommentsButton.classList.remove("hidden");
+      this.endOfCommentsSign.classList.add("hidden");
+
+      this.currentCommentsPageIndex = 0;
+      this.totalNumberOfPagesRequested = 1;
+      this.lastPage = -1;
+      this.requestAndDisplayNewCommentsPage();
+    });
+
+    this.prevCommentsButton.addEventListener('click', () => {
+      this.displayPrevPage();
+    });
+
+    this.nextCommentsButton.addEventListener('click', () => {
+      this.displayNextPage();
+    });
+  }
+
+  hideCommentAtIndex(index) {
+    const commentList = this.comments.children;
+    commentList[index].classList.add("hidden");
+  }
+
+  showCommentAtIndex(index) {
+    const commentList = this.comments.children;
+    commentList[index].classList.remove("hidden");
+  }
+
+  disableButtons() {
+    this.prevCommentsButton.disabled = true;
+    this.nextCommentsButton.disabled = true;
+  }
+
+  enableButtons() {
+    this.prevCommentsButton.disabled = false;
+    this.nextCommentsButton.disabled = false;
+  }
+
+   async displayNextPage() {
+    this.disableButtons();
+
+    this.prevCommentsButton.classList.remove("hidden");
+
+    // this.currentCommentsPageIndex points to the page that is displayed before the user clicks next
+    // if condition checks whether the page to display when user clicks next is the last page, which holds -1 or the last page index that is set in the request/display new comments page function
+    let commentList = this.comments.children;
+    if (this.currentCommentsPageIndex + 1 === this.lastPage) {
+      this.nextCommentsButton.classList.add("hidden");
+      this.endOfCommentsSign.classList.remove("hidden");
+    }
+    // First if condition checks whether the current comments page + 1 (so the requested page) is not the last page.
```

Comment:
Let's see if we can be more succinct:

if (this.currentCommentsPageIndex + 1 < this.totalNumberOfPagesRequested) {
   // There are more pages to view that have already been loaded. Swap the visible comments page.
  ... your code ...
} else {
  // We need to request a new page from the server. The prev/next buttons are already disabled, so awaiting response is safe since the user cannot change the pageIndex while the comments are being fetched.
  .. your code
}

Reply:
Awesome, thank you so much! I combined your two comment suggestions for the else part of the code!